integration/tests/cook/test_basic.py

```diff
@@ -869,6 +869,94 @@ def query():
             # (ensure it still works when there are no live jobs)
             util.kill_groups(self.cook_url, [group_uuid])
 
+    def test_group_change_killed_retries(self):
+        group_spec = self.minimal_group()
+        group_uuid = group_spec['uuid']
+        job_spec = {'group': group_uuid, 'command': f'sleep 1'}
+        try:
+            jobs, resp = util.submit_jobs(self.cook_url, job_spec, 10, groups=[group_spec])
+            self.assertEqual(resp.status_code, 201)
+            # wait for some job to start
+
+            def group_query():
+                return util.group_detail_query(self.cook_url, group_uuid)
+
+            util.wait_until(group_query, util.group_some_job_started)
+            # kill all jobs in the group (and wait for the kill to complete)
+            self.logger.info(f'Killing all jobs in group {group_uuid}.')
+            util.kill_groups(self.cook_url, [group_uuid])
+
+            def jobs_query():
+                return util.query_jobs(self.cook_url, True, job=jobs)
+
+            util.wait_until(jobs_query, util.all_instances_done)
+            # retry all jobs in the group
+            util.retry_jobs(self.cook_url, retries=2, groups=[group_uuid])
+            # wait for some job to start
+            util.wait_until(group_query, util.group_some_job_started)
+            # ensure none of the jobs are still in a failed state
+            job_data = util.query_jobs(self.cook_url, job=jobs)
+            self.assertEqual(200, job_data.status_code)
+            for job in job_data.json():
+                if job['status'] not in ['waiting', 'running']:
+                    job_details = f'Job details: {json.dumps(job, sort_keys=True)}'
+                    self.assertEqual('completed', job['status'], job_details)
+                    self.assertEqual('success', job['state'], job_details)
```

Comment:
Nitpick: I think this would be more clear if we asserted that `job['status']` is either waiting, running, or completed, instead of only having the `assertEquals` if the status is not waiting or running